File: Js/linear_search.js

function linear_search(arrai, target){
    // Return the index position of the target if found, else returns None

    let result = 0; 
    
    for( let i = 0; arrai.length > i; i++){
        if(arrai[i] == target ){
            return i
        }else{
            return null
        }
    }
}

function verify(index){
    if( index != null){
        console.log("Target found at index: ", index );
    }else{
        console.log("Target not found in the List")
    }   
}

let numbers = [1,2,3,4,5,6,7,8,9,10]

let result = linear_search(numbers, 12);
verify(result)

let result2 = linear_search(numbers, 8);
verify(result2)

File: Python/linear_search.py

# Linear Time O(1)
# counting one by one
# the best case runtime on binary search

# Logarithmic Runtime will be dividing the linear Time
# the wors case of runtime will be O(log n)

def linear_search(list, target):
  """
  Return the index position of tghe target if found, else returns None
  """
  # len list es a constan time operation
  for i in range(0, len(list)):
    if list[i] == target: 
      return i
  return None
  

def verify(index):
  if index is not None:
    print("target found at index: ", index)
  else:
    print("Target not found in list")
    

numbers = list(range(0,10))
# print(numbers)

result = linear_search(numbers, 12)
verify(result)
result2 = linear_search(numbers, 9)
verify(result2)

  